x-pack/plugins/ml/public/application/model_management/deployment_params_mapper.ts

```diff
@@ -0,0 +1,229 @@
+    low: { min: 0, max: 2, static: 2 },
+    medium: { min: 3, max: 32, static: 32 },
+    high: { min: 33, max: 99999, static: 100 },
+  };
+
+  /**
+   * vCPUs level breakpoints for serverless projects
+   */
+  private readonly serverlessVCPUBreakpoints: VCPUBreakpoints = {
+    low: { min: 0, max: 2, static: 2 },
+    medium: { min: 3, max: 32, static: 32 },
+    high: { min: 33, max: 500, static: 100 },
+  };
+
+  /**
+   * vCPUs level breakpoints based on the ML server limits.
+   * Either on-prem or cloud with disabled ML autoscaling
+   */
+  private readonly hardwareVCPUBreakpoints: VCPUBreakpoints;
+
+  /**
+   * Result vCPUs level breakpoint based on the cluster env.
+   */
+  private readonly vCpuBreakpoints: VCPUBreakpoints;
+
+  constructor(
+    private readonly modelId: string,
+    private readonly mlServerLimits: MlServerLimits,
+    private readonly cloudInfo: CloudInfo,
+    private readonly showNodeInfo: boolean
+  ) {
+    const maxSingleMlNodeProcessors = this.mlServerLimits.max_single_ml_node_processors;
+
+    this.threadingParamsValues = new Array(THREADS_MAX_EXPONENT)
+      .fill(null)
+      .map((v, i) => Math.pow(2, i))
+      .filter(maxSingleMlNodeProcessors ? (v) => v <= maxSingleMlNodeProcessors : (v) => true);
+
+    const mediumValue = this.mlServerLimits!.total_ml_processors! / 2;
+
+    this.hardwareVCPUBreakpoints = {
+      low: { min: 0, max: 2, static: 2 },
+      medium: { min: Math.min(3, mediumValue), max: mediumValue, static: mediumValue },
+      high: {
+        min: mediumValue + 1,
+        max: this.mlServerLimits!.total_ml_processors!,
+        static: this.mlServerLimits!.total_ml_processors!,
+      },
+    };
+
+    if (!this.showNodeInfo) {
+      this.vCpuBreakpoints = this.serverlessVCPUBreakpoints;
+    } else if (this.cloudInfo.isMlAutoscalingEnabled) {
+      this.vCpuBreakpoints = this.autoscalingVCPUBreakpoints;
+    } else {
+      this.vCpuBreakpoints = this.hardwareVCPUBreakpoints;
+    }
+  }
+
+  private getNumberOfThreads(input: DeploymentParamsUI): number {
+    // 1 thread for ingest at all times
+    if (input.optimized === 'optimizedForIngest') return 1;
+    // for search deployments with low vCPUs level set 2, otherwise max available
+    return input.vCPUUsage === 'low' ? 2 : Math.max(...this.threadingParamsValues);
+  }
+
+  private getAllocationsParams(
+    params: DeploymentParamsUI
+  ): Pick<MlStartTrainedModelDeploymentRequestNew, 'number_of_allocations'> &
+    Pick<
+      Exclude<MlStartTrainedModelDeploymentRequestNew['adaptive_allocations'], undefined>,
+      'min_number_of_allocations' | 'max_number_of_allocations'
+    > {
+    const threadsPerAllocation = this.getNumberOfThreads(params);
+
+    const levelValues = this.vCpuBreakpoints[params.vCPUUsage];
+
+    const maxValue = Math.floor(levelValues.max / threadsPerAllocation) || 1;
+
+    return {
+      number_of_allocations: maxValue,
+      min_number_of_allocations:
```

Comment:
I'm running with `"cluster_name" : "es-only-9.0.0-main-768",` and am seeing this error when attempting to deploy with the 'Low' vCPUs level:

<img width="353" alt="Screenshot 2024-09-26 at 10 52 14" src="https://github.com/user-attachments/assets/57431ee3-7a09-4514-87bc-a7e7ed0e8ea7">

Reply:
Fixed by https://github.com/elastic/kibana/pull/190243/commits/b61ca974a04400cc81dadd307eda18464da73fff

Reply:
Confirmed that the 'Low' level works with the switch back to `1` allocation in https://github.com/elastic/kibana/pull/190243/commits/b61ca974a04400cc81dadd307eda18464da73fff. 

As discussed, we can set this back to `0` (behind a feature flag) when https://github.com/elastic/elasticsearch/pull/113455 is merged.